Core/src/org/sleuthkit/autopsy/textextractors/TextExtractor.java

```diff
@@ -62,23 +50,22 @@ boolean isEnabled() {
      * @see org.sleuthkit.autopsy.textextractors.TextExtractorFactory
      *
      */
-    public abstract Reader getReader() throws ExtractionException;
+    Reader getReader() throws ExtractionException;
 
     /**
      * Determines how the extraction process will proceed given the settings
      * stored in the context instance.
      *
      * @param context Instance containing file config classes
      */
-    void setExtractionSettings(Lookup context) {
+    default void setExtractionSettings(Lookup context) {
         //no-op by default
     }
 
     /**
-     * Exception encountered during
-     * {@link org.sleuthkit.autopsy.textextractors.TextExtractor#getReader()}.
-     * This indicates that there was an internal parsing error that occurred
-     * during the reading of Content text.
+     * Exception encountered during TextExtractor.getReader(). This indicates
+     * that there was an internal parsing error that occurred during the reading
+     * of Content text.
```

Comment:
I can make it more clear. This is a creation error of the Reader itself. For instance, a call to Tika. Or for example the HTMLTextExtractor ... where we concatenate all of the HTML document into a StringBuilder upfront, instead of creating and returning our own Reader implementation.
https://github.com/sleuthkit/autopsy/blob/develop/Core/src/org/sleuthkit/autopsy/textreaders/HtmlTextExtractor.java

Reply:
Clarity in public API is good! :)

Reply:
I'll rename the exception all together then, any ideas?

Reply:
I agree, I can change the name tomorrow and update the docs.